src/graphql.rs

```diff
@@ -316,44 +316,14 @@ fn load_nodes<I, R, N>(
     table: &I,
     after: Option<String>,
     before: Option<String>,
-    mut first: Option<usize>,
-    mut last: Option<usize>,
+    first: Option<usize>,
+    last: Option<usize>,
```

Comment:
first와 last 의 type이 Option<i32> 으로 변경되는 게 올바른 방향으로 보입니다. 변경 부탁드려요.

Reply:
@sophie-cluml  Changing `load_edges` and `load_nodes` from `usize` to `i32` seems to create a lot of errors. There seems to be quite a few functions that rely on first and last to be `usize` for the functions in `graphql.r
s` so when I do change it I might need to change most of the other places to work such as `load` and `tier_to_nodes_with_filter`. Should I continue with the change?

<img width="1920" alt="Screenshot 2024-04-02 at 8 08 27 PM" src="https://github.com/aicers/review-web/assets/102562226/a51b8646-42f7-404f-b735-55c339b227b2">

Reply:
If the change is toward the right direction, the volume of the change is not an issue. We just need to be careful if we are making right changes in each function.
 
For `iter_to_nodes_with_filter` case you mentioned, I would
- maintain the function signature of `iter_to_nodes_with_filter`, so that `len` stays as `usize`.
  -  why : `iter_to_nodes_with_filter` does not want the `len` to be signed int. (Also, length normally are not negative.)
- change the function signature of `load_nodes_with_filter`, by changing `after`, `before`, `first`, `last` to `Option<i32>`  
- make appropriate code change inside `load_nodes_with_filter`, so that the `len` parameter for `iter_to_nodes_with_filter` is still of type `usize`

This discussion can go further, especially on where to draw the line between presentation-layer and logic-layer, and when to use which type. What I have in mind is that `load_nodes_with_filter` is in presentation-layer, so the parameter types are affected by the graphql spec, while `iter_to_nodes_with_filter` lives in logic-layer, so we don't need to care about the graphql spec. But others may have an idea that`iter_to_nodes_with_filter`is also in charge of presentation and bound to graphql spec. But even if that's the case, it seems pretty unusual to change `len` to some signed int.

Hope this comment helps you make decisions along the change. Let me know if you bump into different kind of problems.

Reply:
Ah ok! I think I see where you're going with this.

Reply:
For `first` and `last` of the validation function, should it be `Option<usize>` instead of `Option<i32>`. I read on the docs real quick it should be a number between 1 and 100 so instead of everything turning to `Option<i32>`, is it better to change it to `Option<usize>` or `Option<u32>`? Or should I keep going with `Option<i32>`?

<img width="1920" alt="Screenshot 2024-04-02 at 9 04 03 PM" src="https://github.com/aicers/review-web/assets/102562226/5bfe49c6-9681-4e86-b082-860d8c8f7207">

Reply:
It needs to be `Option<i32>`. GraphQL's `Int` is `i32` in rust.  https://graphql.org/learn/schema/#:~:text=GraphQL%20comes%20with%20a%20set%20of%20default%20scalar%20types%20out%20of%20the%20box%3A

Also, you are not referreing to official graphql's doc.
Take a look at https://graphql.org/learn/pagination/

Reply:
Ah I see! Thank you! I'll continue working with `i32`

Reply:
Done, I changed where there was `Option<usize>` to `Option<i32>` where I thought it would be necessary. Please let me know if there are anything that needs to be changed!